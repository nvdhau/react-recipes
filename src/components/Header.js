import React from 'react';
import { NavLink } from 'react-router-dom';
import PropTypes from 'prop-types';
import Logo from '../static/images/logo.png';

// activeClassName from react-router-dom

const HeaderLink = ({ children, ...props }) => {
  // ...props: paste all props of parent to child
  console.log(props);

  return (
    <NavLink
      exact
      className="p1 mx2 black rounded text-decoration-none"
      activeClassName="bg-white"
      {...props} // is to="/" for Home
    >
      {children}
    </NavLink>
  );
};

HeaderLink.propTypes = {
  children: PropTypes.node,
};

const Header = () => (
  <header className="flex items-center justify-between px4">
    <div>
      <h1 className="h1 inline-block mr2">üçΩ My Recipes</h1>
      <span className="mr1">by</span>
      <img src={Logo} alt="Logo" style={{ width: 30 }} />
    </div>
    <nav>
      {/* Home is children "/" is props */}
      <HeaderLink to="/">Home</HeaderLink>
      <HeaderLink to="/favorites">Favorites</HeaderLink>
      {/* Example NavLink, activeClassName applies when path is matched
      <NavLink
        exact
        to="/"
        className="p1 mx2 black rounded text-decoration-none"
        activeClassName="bg-white"
      >
        Home
      </NavLink> */}
    </nav>
  </header>
);

export default Header;
