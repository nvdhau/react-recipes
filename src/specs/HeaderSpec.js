import React from 'react';
import { BrowserRouter } from 'react-router-dom';
import renderer from 'react-test-renderer';
import Header from '../components/Header';

// think component as IO function
describe('<Header />', () => {
  test('Should render correctly', () => {
    // because Header has NavLink, and NavLink needs to be inside BrowserRouter
    const component = renderer.create(
      <BrowserRouter>
        <Header />
      </BrowserRouter>,
    );
    const tree = component.toJSON();

    expect(tree).toMatchSnapshot();
  });
});
