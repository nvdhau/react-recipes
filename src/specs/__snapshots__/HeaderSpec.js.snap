// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Header /> Should render correctly 1`] = `
<header
  className="flex items-center justify-between px4"
>
  <div>
    <h1
      className="h1 inline-block mr2"
    >
      üçΩ My Recipes
    </h1>
    <span
      className="mr1"
    >
      by
    </span>
    <img
      alt="Logo"
      src="test-file-stub"
      style={
        Object {
          "width": 30,
        }
      }
    />
  </div>
  <nav>
    <a
      aria-current="page"
      className="p1 mx2 black rounded text-decoration-none bg-white"
      href="/"
      onClick={[Function]}
      style={Object {}}
    >
      Home
    </a>
    <a
      aria-current={null}
      className="p1 mx2 black rounded text-decoration-none"
      href="/favorites"
      onClick={[Function]}
    >
      Favorites
    </a>
  </nav>
</header>
`;
